presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -304,14 +302,43 @@ private void finishRowGroup()
         rowGroupRowCount = 0;
     }
 
-    private void writeStripe(FlushReason flushReason)
+    private void flushStripe(FlushReason flushReason)
             throws IOException
     {
-        if (stripeRowCount == 0) {
-            return;
+        List<DataOutput> outputData = new ArrayList<>();
+        long stripeStartOffset = output.longSize();
+        // add header to first stripe (this is not required but nice to have)
+        if (closedStripes.isEmpty()) {
+            outputData.add(createDataOutput(MAGIC));
```

Comment:
Maybe create a helper function for these two lines similar to `bufferStripeData` and `bufferFileFooter` for readability.